src/App.tsx

```diff
@@ -15,36 +19,95 @@
   'Garlic',
 ];
 
+
+function sorteGoods(goods: string[], sortedField:string, reversed:boolean): string[] {
+  goods.sort((a, b) => {
+    switch (sortedField) {
+      case SORT_BY_ALPHABET:
+        return a.localeCompare(b);
+      case SORT_BY_LENGTH:
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (reversed) {
+    goods.reverse();
+  }
+
+  return goods;
+}
+
+
 export const App: React.FC = () => {
+  const [sortedField, setSortedField] = useState('');
+  const [reversed, setReversed] = useState(false);
+
+  let goods = sorteGoods([...goodsFromServer], sortedField, reversed);
+
+  function resetSortedGoods() {
+    setSortedField('');
+    setReversed(false);
+    goods = [...goodsFromServer];
```

Comment:
Directly assigning `goods` in `resetSortedGoods` will not trigger a re-render. Consider using state to manage `goods` to ensure the component updates correctly when the goods list is reset.

Reply:
review it again